crates/flipperzero/src/dialogs/mod.rs

```diff
@@ -164,6 +202,84 @@ impl DialogMessageButton {
     }
 }
 
+impl<'a> DialogFileBrowserOptions<'a> {
+    /// Creates a new dialog file browser options and initializes to default values.
+    pub fn new(extension: Option<&'a CStr>, icon: Option<&'a sys::Icon>) -> Self {
+        let mut dialog_file_browser_options = Self {
+            data: unsafe { mem::zeroed() },
+            _phantom: PhantomData,
+        };
+        let ext_ptr = extension
+            .unwrap_or(Default::default())
+            .as_ptr();
+        let icon_ptr = match icon {
+            Some(icon) => icon as *const sys::Icon,
+            None => ptr::null(),
+        };
+        unsafe {
+            sys::dialog_file_browser_set_basic_options(
+                &mut dialog_file_browser_options.data as *mut sys::DialogsFileBrowserOptions,
```

Comment:
This shouldn't be moving the address of `dialog_file_browser_options` we are passing a raw pointer to the inner `sys::DialogsFileBrowserOptions` for the C function to populate. This is just a helper function that sets things like `options->icon = icon;` in C instead of us assigning the variable directly in Rust. Anyways, I'm removing this FFI call and now doing all the default value assignments in Rust.

Reply:
Actually, we can use `dialog_file_browser_set_basic_options` safely (from API standpoint) which would allow us to not copy the defaults (which may be important if the deafult change in C impl) and yet keep soundness: you could use `MaybeUninit` for stack-allocating the structure and then passing it to `dialog_file_browser_set_basic_options`.

Reply:
Ah, well now that you've seen things both ways do you have a strong preference? I also just noticed that `dialog_file_browser_set_basic_options()` doesn't touch the `hide_dot_files` member. It seems to rely on a struct being zeroed and zero being a falsey value in C.

Reply:
That's a really interesting question.

It may just be secretly unsound in flipperzero-firmware since their `DialogsFileBrowserOptions browser_options` initialization does not guarantee zero-filling (AFAIK).

Let's keep rust-based initialization as for now, since it is easier to speculate on its safety as for now. But I will ask firmware maintainers if it's just an overlooked field in their initialization and we can ask them to make `dialog_file_browser_set_basic_options()` always fully initialize the instance so that we can rely on it (and in this case we will add a future PR to replace current approach with `MaybeUnit`-based).

Reply:
I've [asked](https://t.me/flipperzero/103925/177467) them but I suggest that we don't wait fr the response (since it also requires filing a PR to firmware, a new release of firware and update of our bindings) and use Rust-approach by now.